src/vsr/replica.zig

```diff
@@ -1454,10 +1465,15 @@ pub fn ReplicaType(
             assert(message.header.operation != .root);
             assert(message.header.view <= self.view); // The client's view may be behind ours.
 
-            const realtime = self.clock.realtime_synchronized() orelse {
-                log.err("{}: on_request: dropping (clock not synchronized)", .{self.replica});
-                return;
-            };
+            // Messages with `client == 0` are sent from itself,
+            // it's not needed to get a different timestamp.
+            const realtime: i64 = if (message.header.client == 0)
+                @intCast(message.header.timestamp)
```

Comment:
Checking my understanding -- the header timestamp here is only non-zero for AOF recovery, right?

Reply:
Yes, it relies on this code to use the prepare timestamp (as desired)
However, the AOF already has a specific block to this.

https://github.com/tigerbeetle/tigerbeetle/blob/4fa955a01e66eda74ddec42375cf5911e9ba2728/src/vsr/replica.zig#L5592-L5600